src/main/java/com/linecorp/armeria/server/metrics/ServiceMetricConsumer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2015 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.metrics;
+
+import java.util.Objects;
+
+@FunctionalInterface
+public interface ServiceMetricConsumer {
+    /**
+     * Invoked for each request that has been processed
+     *
+     * @param status flag whether it's successful or not
+     * @param processTimeNanos elapsed nano time processing request
+     * @param requestSize number of bytes in request if possible, otherwise it will be 0
+     * @param responseSize number of bytes in response if possible, otherwise it will be 0
+     */
+    void onComplete(Status status, long processTimeNanos, int requestSize, int responseSize);
+
+    default ServiceMetricConsumer andThen(ServiceMetricConsumer other) {
+        Objects.requireNonNull(other);
+        return (status, processNanoTime, requestSize, responseSize) -> {
+            onComplete(status, processNanoTime, requestSize, responseSize);
+            other.onComplete(status, processNanoTime, requestSize, responseSize);
```

Comment:
Perhaps we could catch a `Throwable` and log when `onComplete()` ever raises an exception, so that the next consumer isn't affected by it?